mlir/test/Dialect/SCF/wrap-while-loop-in-zero-trip-check.mlir

```diff
@@ -20,7 +20,7 @@ func.func @wrap_while_loop_in_zero_trip_check(%bound : i32) -> i32 {
 // CHECK-SAME:      %[[BOUND:.*]]: i32) -> i32 {
 // CHECK-DAG:     %[[C0:.*]] = arith.constant 0 : i32
 // CHECK-DAG:     %[[C5:.*]] = arith.constant 5 : i32
-// CHECK-DAG:     %[[PRE_COND:.*]] = arith.cmpi slt, %[[C0]], %[[BOUND]] : i32
+// CHECK-DAG:     %[[PRE_COND:.*]] = arith.cmpi sgt, %[[BOUND]], %[[C0]] : i32
```

Comment:
Actually I don't quite get this change: the condition `(c0 < bound)` isn't the same as `(bound < c0)`?

Reply:
`arith.cmpi` folding moves constants to the RHS :)

Reply:
Do you know why is your change triggering this?

But also I'm still unsure why this folding is correct?

Reply:
`c0 < bound iff bound > c0` works. If we were using `sge` instead, we'd end up with `bound >= c0`. This introduces the case in which `bound = c0`, which is not in the original `c0 < bound` precondition. The folding pattern is correct.

Reply:
Or right somehow I had in mind to provide the reverse condition, not the same condition with the reversed arguments... Alright time to go to bed!

Reply:
It happened to me for a couple of minutes too! This can be tricky sometimes. Thanks!